### PR TITLE
Automatically apply CNI on multinode clusters

### LINKED ISSUE #7459: Figure out automated CNI for multi-node clusters
Currently multinode clusters need a CNI applied manually to make `minikube service` work. We should decide what CNI will be the canonical CNI choice to apply automatically for multinode and actually apply it on start or add node.

### PR DESCRIPTION
**Pull request by sharifelgamal**
Fixes #7459

### CODE REVIEW

File: pkg/minikube/bootstrapper/kubeadm/kubeadm.go [hunk 1]
```diff
@@ -811,6 +811,7 @@ func kubectlPath(cfg config.ClusterConfig) string {
 
 // applyKICOverlay applies the CNI plugin needed to make kic work
 func (k *Bootstrapper) applyKICOverlay(cfg config.ClusterConfig) error {
+	fmt.Println("APPLY KIC OVERLAY")
```
Comment (medyagh) [inline pkg/minikube/bootstrapper/kubeadm/kubeadm.go:814]: remove debugging.

File: pkg/minikube/driver/driver.go [hunk 2]
```diff
@@ -163,9 +162,9 @@ func FlagDefaults(name string) FlagHints {
 	fh := FlagHints{}
 	if name != None {
 		fh.CacheImages = true
-		// only for kic, till other run-times are available we auto-set containerd.
+		// only for kic, until other runtimes are available we auto-set containerd.
```
Comment (medyagh) [inline pkg/minikube/driver/driver.go:165]: this comment actually is no long true, we have other run-time,s we can just delete this comment.

File: pkg/minikube/config/config.go [hunk 3]
```diff
@@ -200,3 +201,29 @@ func (c *simpleConfigLoader) WriteConfigToFile(profileName string, cc *ClusterCo
...
+		cc.KubernetesConfig.NetworkPlugin = "cni"
+		if err := cc.KubernetesConfig.ExtraOptions.Set(fmt.Sprintf("kubeadm.pod-network-cidr=%s", DefaultPodCIDR)); err != nil {
+			return err
+		}
+		return SaveProfile(cc.Name, cc)
+	}
+	return nil
+}
+
+// MultiNode returns true if the cluster
```
Comment (medyagh) [inline pkg/minikube/config/config.go:218]: finish comment... :)

File: pkg/minikube/config/config.go [hunk 4]
```diff
@@ -200,3 +201,29 @@ func (c *simpleConfigLoader) WriteConfigToFile(profileName string, cc *ClusterCo
 	}
 	return ioutil.WriteFile(path, contents, 0644)
 }
+
+// MultiNodeCNIConfig add default CNI config needed for multinode clusters and saves off the config
+func MultiNodeCNIConfig(cc *ClusterConfig) error {
+	if cc.KubernetesConfig.ExtraOptions.Get("pod-network-cidr", "kubeadm") == "" {
+		fmt.Println("SETTING CNI CONFIG")
```
Comment (medyagh) [inline pkg/minikube/config/config.go:208]: remove deubbing there too.

File: pkg/minikube/config/config.go [hunk 5]
```diff
@@ -200,3 +201,29 @@ func (c *simpleConfigLoader) WriteConfigToFile(profileName string, cc *ClusterCo
 	}
 	return ioutil.WriteFile(path, contents, 0644)
 }
+
+// MultiNodeCNIConfig add default CNI config needed for multinode clusters and saves off the config
+func MultiNodeCNIConfig(cc *ClusterConfig) error {
+	if cc.KubernetesConfig.ExtraOptions.Get("pod-network-cidr", "kubeadm") == "" {
```
Comment (medyagh) [inline pkg/minikube/config/config.go:207]: what is this line doing ? is it getting two options at same time and check if they are "" ?

Comment (tstromberg) [reply pkg/minikube/config/config.go:207]: ExtraOptions is multi-level: this returns the value of `pod-network-cidr` within the `kubeadm` component.

**Pull request merged**